lib/whitehall_importer/integrity_checker/unpublishing_check.rb

```diff
@@ -0,0 +1,59 @@
+module WhitehallImporter
+  class IntegrityChecker::UnpublishingCheck
+    attr_reader :edition, :publishing_api_unpublishing
+
+    def initialize(edition, publishing_api_unpublishing)
+      @edition = edition
+      @publishing_api_unpublishing = publishing_api_unpublishing
+    end
+
+    def expected_type?
+      withdrawal_type_matches? || removed_type_matches?
+    end
+
+    def expected_unpublishing_time?
+      return true unless edition.withdrawn?
+
+      unpublishing_time_matches?
+    end
+
+    def expected_alternative_path?
+      return true unless edition.removed?
+
+      alternative_path_matches?
+    end
+
+    def expected_explanation?
+      Sanitize.clean(publishing_api_unpublishing["explanation"]).squish ==
+        Sanitize.clean(edition_explanation_html).squish
+    end
+
+  private
+
+    def withdrawal_type_matches?
+      publishing_api_unpublishing["type"] == "withdrawal" && edition.withdrawn?
+    end
+
+    def removed_type_matches?
+      publishing_api_type = publishing_api_unpublishing["type"]
+      %w[gone redirect].include?(publishing_api_type) && edition.removed?
```

Comment:
Should we be checking that redirect and gone match too? maybe like:

```
case publishing_api_unpublishing["type"]
when "gone"
  edition.removed? && !edition.status.details.redirect?
when "redirect"
  edition.removed? && edition.status.details.redirect?
else
  false
end
```

You could also put withdrawal in too and have it covered by a single method

Reply:
I'm not sure that it matters that much, we don't distinguish between 'gone' and 'redirect' in Content Publusher, and 'gone' items can have a redirect in Publishing API?

Reply:
We do distinguish in CP with the redirect on the Removal object.

In Publishing API a gone with an alternative path isn't a redirect it'll be a gone with a link shown on the page. For example: https://www.gov.uk/government/statistics/higher-education-provider-data-finance-2018-to-2019

Reply:
I'm now checking for gone and redirect separately.